src/mobile-app/hooks/use-runs.ts

```diff
@@ -1,46 +1,61 @@
 import { IExperiment, IExperimentData } from "../../shared/experiment-types";
 import { useLocalStorage } from "./use-local-storage";
+import { useState } from "react";
 
 export interface IRun {
   key: string;
+  experimentIdx: number;
   experiment: IExperiment;
   data: IExperimentData;
 }
 
 export interface IUseRunsResult {
   runs: IRun[];
-  addNewRun: (experiment: IExperiment) => IRun;
-  saveRunData: (runKey: string, data: IExperimentData) => void;
+  startNewRun: (experiment: IExperiment) => IRun;
+  activeRun: IRun | null;
+  setActiveRun: (run: IRun | null) => void;
+  saveActiveRunData: (data: IExperimentData) => void;
   resetRuns: () => void;
 }
 
 export const useRuns = (): IUseRunsResult => {
   const [runs, setRuns] = useLocalStorage<IRun[]>("runs", []);
+  const [ activeRun, setActiveRun ] = useState<IRun | null>(null);
 
-  const addNewRun = (experiment: IExperiment) => {
+  const startNewRun = (experiment: IExperiment) => {
     const timestamp = Date.now();
+    let experimentIdx = 1;
+    runs.forEach(run => {
+      if (run.experiment.metadata.name === experiment.metadata.name) {
+        experimentIdx += 1;
+      }
+    });
```

Comment:
this could also be `const experimentIdx = runs.filter(run => run.experiment.metadata.name === experiment.metadata.name).length + 1;`

Reply:
Thanks, I'll change that in the upcoming PRs.